source/common/formatter/substitution_formatter.cc

```diff
@@ -1294,9 +1294,9 @@ ProtobufWkt::Value FilterStateFormatter::formatValue(const Http::RequestHeaderMa
   }
 
   ProtobufWkt::Value val;
-  try {
-    MessageUtil::jsonConvertValue(*proto, val);
-  } catch (EnvoyException& ex) {
+  TRY_ASSERT_MAIN_THREAD { MessageUtil::jsonConvertValue(*proto, val); }
```

Comment:
Pretty sure this is on workers?

Reply:
Doesn't break any test when I add the main thread assertion. maybe because of missing test coverage, I will take a look at the test coverage later.

Reply:
Thanks, I will see whether some integration test can be added.